Control/Ardupilot: Various Ardupilot fixes

krisklau opened the pull request:
Concerning "Control/UAV/Ardupilot: Force setup of rates if lost connection due to timeout.", is very useful when the pixhawk is restarted but not Dune (might happen during battery change)

krisklau commented:
Me an @jffortuna: OK for merge (we will do it later)

mariacosta commented:
Hi,

I noticed on the dispatch z and heading commit you have repeated code, i.e.
both on the *if (m_vehicle_type == VEHICLE_COPTER)* and the respective
*else *you have exactly the same content.


Best regards,
Maria Costa

2016-04-14 11:57 GMT+01:00 krisklau <notifications@github.com>:

> Concerning "Control/UAV/Ardupilot: Force setup of rates if lost connection
> due to timeout.", is very useful when the pixhawk is restarted but not Dune
> (might happen during battery change)
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/LSTS/dune/pull/57
> Commit Summary
>
>    - Control/UAV/Ardupilot: Only dispatch desired z and heading from
>    ardupilot when ardupilot tracker is enabled.
>    - Control/UAV/Ardupilot: Handle copter modes Stabilize and Land.
>    - Control/UAV/Ardupilot: Force setup of rates if lost connection due
>    to timeout.
>
> File Changes
>
>    - *M* src/Control/UAV/Ardupilot/Task.cpp
>    <https://github.com/LSTS/dune/pull/57/files#diff-0> (38)
>
> Patch Links:
>
>    - https://github.com/LSTS/dune/pull/57.patch
>    - https://github.com/LSTS/dune/pull/57.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/LSTS/dune/pull/57>
>

krisklau commented:
Rebased on master. Removed the repeated code you mentioned @mariacosta, the reason it was there in the first place was for another expansion which needed different behavior for each vehicle. But this is not needed now.
